test/libsolidity/smtCheckerTests/special/msg_value_3.sol

```diff
@@ -0,0 +1,13 @@
+pragma experimental SMTChecker;
+contract C {
+	bool lock = true;
+	function f() public {
+		lock = false;
+		g();
+		lock = true;
+	}
+	function g() public payable {
+		require(lock == false);
+		assert(msg.value == 0);
```

Comment:
Shouldn't CHC say that the assertion is redundant?

Reply:
Not sure what you mean. For assertions it either reports `assertion violation here` or nothing, in case it can prove that the assertion is true, which is the case here.

Reply:
Ah, nice.